src/controllers/user/user.controller.ts

```diff
@@ -1,119 +1,79 @@
+            await userService.updateUser(_id, {photo_path: `${photoDir}/${photoName}`});
+        }
 
-      fs.mkdirSync(resolvePath(`${appRoot}/static/${photoDir}`), {recursive: true});
-      await userPhoto.mv(resolvePath(`${appRoot}/static/${photoDir}/${photoName}`));
-      await userService.updateUser(_id, {photo_path: `${photoDir}/${photoName}`});
+        res.status(ResponseStatusCodesEnum.CREATED).end();
     }
 
-    res.status(ResponseStatusCodesEnum.CREATED).end();
-  }
-
-  getUserInfoByToken(req: IRequestExtended, res: Response, next: NextFunction) {
-
-    const {_id, email, phone_number, name, surname, role_id, status_id, photo_path, groups_id} = req.user as IUser;
-
-    const user: IUserSubjectModel = {
-      _id,
-      email,
-      phone_number,
-      name,
-      surname,
-      role_id,
-      status_id,
-      photo_path,
-      groups_id
-    };
+    getUserInfoByToken(req: IRequestExtended, res: Response, next: NextFunction) {
 
-    res.json(user);
-  }
+        const {_id, email, phone_number, name, surname, role_id, status_id, photo_path, groups_id} = req.user as IUser;
 
-  async updateUserByID(req: IRequestExtended, res: Response, next: NextFunction) {
-    const {user_id} = req.params;
-    const updateInfo = req.body as IUser;
-    const appRoot = (global as any).appRoot;
-    const [userPhoto] = req.photos as UploadedFile[];
+        const user: IUserSubjectModel = {
+            _id,
+            email,
+            phone_number,
+            name,
+            surname,
+            role_id,
+            status_id,
+            photo_path,
+            groups_id
+        };
 
-    if (userPhoto) {
-
-      const photoDir = `user/${user_id}/photo`;
+        res.json(user);
+    }
 
-      const photoExtension = userPhoto.name.split('.').pop();
-      const photoName = `${uuid.v1()}.${photoExtension}`;
-      const oldPhotos = fs.readdirSync(resolvePath(`${appRoot}/static/${photoDir}`));
+    async updateUserByID(req: IRequestExtended, res: Response, next: NextFunction) {
+        const {user_id} = req.params;
+        let updateInfo = req.body as IUser;
 
-      if (oldPhotos) {
+        const appRoot = (global as any).appRoot;
+        const [userPhoto] = req.photos as UploadedFile[];
 
-        for (const photo of oldPhotos) {
-          fs.unlinkSync(resolvePath(`${appRoot}/static/${photoDir}/${photo}`));
+        if (userPhoto) {
+            updateInfo = await updatedUserPhotoMv(user_id, userPhoto, updateInfo, appRoot);
         }
-      }
 
-      fs.mkdirSync(resolvePath(`${appRoot}/static/${photoDir}`), {recursive: true});
-
-      await userPhoto.mv(resolvePath(`${appRoot}/static/${photoDir}/${photoName}`));
-      updateInfo.photo_path = `${photoDir}/${photoName}`;
-    }
+        await userService.updateUser(user_id, updateInfo);
 
-    const updateValidity = Joi.validate(updateInfo, updateDataValidator);
+        const user = await userService.getUserByID(user_id);
```

Comment:
Will be better to use ```js {new: true} ``` in service (line 59) and don't find user again here. 
But you should remove the password. You can show how we did it in `getUserById` method or use your own logic (delete user, password);

```js 
------>SERVICE<-------

  updateUser(user_id: string, patchObject: Partial<IUser>): Promise<any> {
    const UserModel = model<UserType>(DatabaseTablesEnum.USER_COLLECTION_NAME, UserSchema);

    return UserModel
      .findByIdAndUpdate(user_id, patchObject, {new: true}) as any;
  }



----->CONTROLLER<-----

const updatedUser = await userService.updateUser(user_id, updateInfo);
// REMOVE PASSWORD

res.json({data: updatedUser});
```

Reply:
Done